base/sparse/sparsematrix.jl

```diff
@@ -1504,148 +1504,281 @@ end
     sec, sech, secd, asech,
     csc, csch, cscd, acsch)
 
-
-## Broadcasting kernels specialized for returning a SparseMatrixCSC
-
-# Operations with zero result if both operands are zero
-function gen_broadcast_body_sparse(f::Function, is_first_sparse::Bool)
-    F = Expr(:quote, f)
-    quote
-        Base.Broadcast.check_broadcast_indices(indices(B), A_1)
-        Base.Broadcast.check_broadcast_indices(indices(B), A_2)
-
-        colptrB = B.colptr; rowvalB = B.rowval; nzvalB = B.nzval
-        colptr1 = A_1.colptr; rowval1 = A_1.rowval; nzval1 = A_1.nzval
-        colptr2 = A_2.colptr; rowval2 = A_2.rowval; nzval2 = A_2.nzval
-
-        nnzB = isempty(B) ? 0 : (nnz(A_1) * div(B.n, A_1.n) * div(B.m, A_1.m)  +
-                                 nnz(A_2) * div(B.n, A_2.n) * div(B.m, A_2.m))
-        if length(rowvalB) < nnzB
-            resize!(rowvalB, nnzB)
-        end
-        if length(nzvalB) < nnzB
-            resize!(nzvalB, nnzB)
-        end
-        z = zero(Tv)
-
-        ptrB = 1
-        colptrB[1] = 1
-
-        Tr1 = eltype(rowval1)
-        Tr2 = eltype(rowval2)
-
-        @inbounds for col = 1:B.n
-            ptr1::Int  = A_1.n == 1 ? colptr1[1] : colptr1[col]
-            stop1::Int = A_1.n == 1 ? colptr1[2] : colptr1[col+1]
-            ptr2::Int  = A_2.n == 1 ? colptr2[1] : colptr2[col]
-            stop2::Int = A_2.n == 1 ? colptr2[2] : colptr2[col+1]
-
-            if  A_1.m ==  A_2.m || (A_1.m == 1 && ptr1 == stop1) || (A_2.m == 1 && ptr2 == stop2)
-                while ptr1 < stop1 && ptr2 < stop2
-                    row1 = rowval1[ptr1]
-                    row2 = rowval2[ptr2]
-                    if row1 < row2
-                        res = ($F)(nzval1[ptr1], z)
-                        if res != z
-                            rowvalB[ptrB] = row1
-                            nzvalB[ptrB] = res
-                            ptrB += 1
-                        end
-                        ptr1 += 1
-                    elseif row2 < row1
-                        res = ($F)(z, nzval2[ptr2])
-                        if res != z
-                            rowvalB[ptrB] = row2
-                            nzvalB[ptrB] = res
-                            ptrB += 1
-                        end
-                        ptr2 += 1
-                    else
-                        res = ($F)(nzval1[ptr1], nzval2[ptr2])
-                        if res != z
-                            rowvalB[ptrB] = row1
-                            nzvalB[ptrB] = res
-                            ptrB += 1
-                        end
-                        ptr1 += 1
-                        ptr2 += 1
-                    end
+## Broadcast operations involving two sparse matrices
+function broadcast{Tf}(f::Tf, A::SparseMatrixCSC, B::SparseMatrixCSC)
+    indextypeC = promote_type(eltype(A.rowval), eltype(B.rowval))
+    entrytypeC = Base.Broadcast.promote_eltype_op(f, A, B)
```

Comment:
`promote_eltype_op` is not exported, necessitating the `Base.Broadcast.` qualification? Thanks!